hedged.go

```diff
@@ -53,40 +53,49 @@ func (ht *hedgedTransport) RoundTrip(req *http.Request) (*http.Response, error)
 	defer cancel()
 	req = req.WithContext(ctx)
 
-	var res interface{}
-	resultCh := make(chan interface{}, ht.upto)
+	var res *http.Response
+	var err error
+	resultCh := make(chan *http.Response, ht.upto)
+	errorCh := make(chan error, ht.upto)
 
+Loop:
 	for sent := 0; ; sent++ {
 		if sent < ht.upto {
 			runInPool(func() {
 				resp, err := ht.rt.RoundTrip(req)
 				if err != nil {
-					resultCh <- err
+					errorCh <- err
 				} else {
 					resultCh <- resp
 				}
 			})
 		}
+		// try to read result channel first, before blocking on all other channels
+		select {
+		case res = <-resultCh:
+			break Loop
+		default:
+		}
 
 		select {
 		case res = <-resultCh:
+			break Loop
+		case err = <-errorCh:
```

Comment:
I don't know how bad it's, but we're loosing 1st error, probably we should keep it, wdyt?

(best case is to create a multi-error, but that's too much :)

Reply:
Yes, we are going to convey only the last error to the user, not sure if we want to do something more sophisticated than that

Reply:
I added support for multi-errors